### DISCUSSION TITLE
Why StackLayout has to be split into 3 different controls?

### DISCUSSION THREAD

**Discussion by pierre01** [dotnet/maui#6781] [General] [answered]
### Description

StackLayout is a control that has 2 other controls HorizontalStackLayout and VerticalStackLayout that can be created using the Stacklayout control with the property **Orientation**.

I personally don't mind the redundancy in syntax but after reading the documentation I read that **HorizontalStackLayout and VerticalStackLayout  are different in performance.** from StackLayout ... WHY?

### Public API Changes

StackLayout could be used for both HorizontalStackLayout and VerticalStackLayout   through the compiler with the Orientation property.

I am not opposed to the redundancy / shorthand with HorizontalStackLayout and VerticalStackLayout
but **I don't understand the performance differences mentioned in the documentation**

### Intended Use-Case

<StackLayout Orientation="Horizontal"....
Should be the same as 
<HorizontalStackLayout....

<StackLayout Orientation="Vertical"....
Should be the same as 
<VerticalStackLayout....

**Comment by hartez** [comment, marked as answer]
> HorizontalStackLayout and VerticalStackLayout are different in performance. from StackLayout ... WHY?

Because StackLayout has to handle the `Orientation` property, and the other two do not. 

In MAUI.Controls, the StackLayout is built from the Horizontal- and VerticalStackLayout classes - it looks at the value of the `Orientation` property and switches between the other two classes internally. `Orientation` is a bindable property, which means the StackLayout has to be prepared for it change at any time, and update the internal layout accordingly. And this has a slight performance impact. 

So if you know that your layout orientation will never change, it's very slightly more performant to use a VerticalStackLayout or HorizontalStackLayout. That's the only performance difference. It's small, but it's worth noting because it _can_ make a difference at scale (for instance, inside of DataTemplates).

We also provide the StackLayout with the `Orientation` property to make it easier to migrate from Forms, which _only_ had StackLayout.

**Reply by J-Swift** [reply]
The Vertical and Horizontal versions also dont support `_AndExpand` right? I believe `StackLayout` preserves that API (and I've said before, but my opinion is that this should not be removed)

**Reply by hartez** [reply]
That is correct, the HorizontalStackLayout and the VerticalStackLayout do not support the deprecated "AndExpand" layout options. StackLayout does (for convenience of migration from Forms). 

That support also has a small performance cost, so if you don't need it then using VerticalStackLayout/HorizontalStackLayout will avoid that cost.

**Reply by pierre01** [reply]
Ok That makes sense now. Thank you guys.
